### PR TITLE
refactor(manager/jenkins): correct type-check for version

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes

Add `number` as a possible type for Jenkins plugin version

## Context

Previously, the version property could be not-defined, or string. Therefore the `if (plugin.source?.version) {` narrowed the type to `string`, but that doesn't reflect reality where the type can be a `number` as well

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by JamieMagee** [comment]
@agaudreault-jive could you check this please if you have some time? You added this check initially, and I want to be sure I'm not misunderstanding your intention, or missing another case.

File: lib/modules/manager/jenkins/extract.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ function getDependency(plugin: JenkinsPlugin): PackageDependency {
 
   if (plugin.source?.version) {
     dep.currentValue = plugin.source.version.toString();
-    if (typeof plugin.source.version !== 'string') {
+    if (!is.string(plugin.source.version)) {
```
Comment (viceice) [inline lib/modules/manager/jenkins/extract.ts:22]: is there a test for number?

why it can be number? 🤔

**Comment by agaudreault-jive** [comment]
It's been a while, but if I remember correctly it's because if you write "2.0" in your plugins.yaml, the parser will read it as a number instead of a string. From there, it is impossible to know of it was "2.0" or "2" and it was causing issue during the comparison. 

The fix for version using this pattern is to update the jenkins plugins.yaml file and add quotes to the version, forcing it to be a string.

**Comment by JamieMagee** [comment]
Yeah, the example you included was `2.10`, which is incorrectly read as `2.1`:

https://github.com/renovatebot/renovate/blob/9d1994eabb8f5ac1d9f62e2c42f41226d4e03871/lib/modules/manager/jenkins/__fixtures__/plugins.yaml#L8-L10

@viceice I guess `unknown` would be a truly accurate type when we're reading untrusted input. I think `string | number` is just to catch cases where people accidentally put a number where a string is required.

**Comment by viceice** [comment]
we should warn if it's a number

**Comment by JamieMagee** [comment]
Do the existing check and warning not cover this already?

https://github.com/renovatebot/renovate/blob/f3c00aa39c9dd2ea1d927440e0bff29c7d2d9c83/lib/modules/manager/jenkins/extract.ts#L22-L28

**Review by viceice — APPROVED** [review]

**Pull request merged**